netmaster/daemon/daemon.go

```diff
@@ -88,18 +91,26 @@ func (d *MasterDaemon) Init() {
 }
 
 func (d *MasterDaemon) registerService() {
-	// Get the address to be used for local communication
-	localIP, err := getLocalAddr()
-	if err != nil {
-		log.Fatalf("Error getting local IP address. Err: %v", err)
+	var err error
+
+	ctrlURL := strings.Split(d.ControlURL, ":")
+	masterIP := ctrlURL[0]
+	masterPort, _ := strconv.Atoi(ctrlURL[1])
+
+	if len(masterIP) == 0 {
```

Comment:
Would be nice to perform this check when we parse this command line and init d.ControlURL there.